helper.go

```diff
@@ -31,12 +30,30 @@ func decodeTimestamp(uuidBytes []byte) uint64 {
 
 // Helper function to parse and sanitize a UUID string.
 func parseUUID(uuid string) ([]byte, error) {
-	uuid = strings.ReplaceAll(uuid, "-", "")
-	if len(uuid) != 32 {
+	if len(uuid) == 32 {
+		// Fast path for UUIDs without dashes
+		return hex.DecodeString(uuid)
+	} else if len(uuid) == 36 {
+		// Validate dash positions
+		if uuid[8] != '-' || uuid[13] != '-' || uuid[18] != '-' || uuid[23] != '-' {
+			return nil, errors.New("invalid UUID format")
+		}
+	} else {
 		return nil, errors.New("invalid UUID length")
 	}
```

Comment:
```suggestion
	switch len(uuid) {
		case 32:
			// Fast path for UUIDs without dashes
			return hex.DecodeString(uuid)
		case 36:
			// Validate dash positions
			if uuid[8] != '-' || uuid[13] != '-' || uuid[18] != '-' || uuid[23] != '-' {
				return nil, errors.New("invalid UUID format")
			}
		default:
	 		return nil, errors.New("invalid UUID length")
 	}
```

Reply:
Also, if you consider this

https://github.com/ash3in/uuidv8/pull/12#discussion_r1896858309

```suggestion
	switch len(uuid) {
		case 32:
			// Fast path for UUIDs without dashes
			return hex.DecodeString(uuid)
		case 36:
			// Validate dash positions
			if uuid[8] != '-' || uuid[13] != '-' || uuid[18] != '-' || uuid[23] != '-' {
				return nil, errors.New("invalid UUID format")
			}
			return hex.DecodeString(strings.ReplaceAll(string(uuid), "-", ""))
		default:
	 		return nil, errors.New("invalid UUID length")
 	}
```

Reply:
Or this

```suggestion
	 if len(uuid) == 36 {
		// Validate dash positions
		if uuid[8] != '-' || uuid[13] != '-' || uuid[18] != '-' || uuid[23] != '-' {
			return nil, errors.New("invalid UUID format")
	}
	if len(uuid) != 32 {
 		return nil, errors.New("invalid UUID length")
 	}
	// Fast path for UUIDs without dashes
	return hex.DecodeString(uuid)
```